test/unit/SwapExactOut.t.sol

```diff
@@ -0,0 +1,350 @@
+        _swapExactOutTest(dai, sDai, 80e18, 100e18, swapper, swapper);
+    }
+
+    function test_swapExactOut_daiToUsdc_differentReceiver() public assertAtomicPsmValueDoesNotChange {
+        _swapExactOutTest(dai, usdc, 100e6, 100e18, swapper, receiver);
+    }
+
+    function test_swapExactOut_daiToSDai_differentReceiver() public assertAtomicPsmValueDoesNotChange {
+        _swapExactOutTest(dai, sDai, 80e18, 100e18, swapper, receiver);
+    }
+
+    function testFuzz_swapExactOut_daiToUsdc(
+        uint256 amountOut,
+        address fuzzSwapper,
+        address fuzzReceiver
+    ) public {
+        vm.assume(fuzzSwapper  != address(psm));
+        vm.assume(fuzzReceiver != address(psm));
+        vm.assume(fuzzReceiver != address(0));
+
+        amountOut = _bound(amountOut, 1, USDC_TOKEN_MAX);  // Zero amount reverts
+        uint256 amountIn = amountOut * 1e12;
+        _swapExactOutTest(dai, usdc, amountOut, amountIn, fuzzSwapper, fuzzReceiver);
+    }
+
+    function testFuzz_swapExactOut_daiToSDai(
+        uint256 amountOut,
+        uint256 conversionRate,
+        address fuzzSwapper,
+        address fuzzReceiver
+    ) public {
+        vm.assume(fuzzSwapper  != address(psm));
+        vm.assume(fuzzReceiver != address(psm));
+        vm.assume(fuzzReceiver != address(0));
+
+        amountOut      = _bound(amountOut,      1,       DAI_TOKEN_MAX);
+        conversionRate = _bound(conversionRate, 0.01e27, 100e27);  // 1% to 10,000% conversion rate
+        mockRateProvider.__setConversionRate(conversionRate);
+
+        uint256 amountIn = amountOut * conversionRate / 1e27;
+
+        _swapExactOutTest(dai, sDai, amountOut, amountIn, fuzzSwapper, fuzzReceiver);
+    }
+
+}
+
+contract PSMSwapExactOutUsdcAssetInTests is PSMSwapExactOutSuccessTestsBase {
+
+    function test_swapExactOut_usdcToDai_sameReceiver() public assertAtomicPsmValueDoesNotChange {
+        _swapExactOutTest(usdc, dai, 100e18, 100e6, swapper, swapper);
+    }
+
+    function test_swapExactOut_usdcToSDai_sameReceiver() public assertAtomicPsmValueDoesNotChange {
+        _swapExactOutTest(usdc, sDai, 80e18, 100e6, swapper, swapper);
+    }
+
+    function test_swapExactOut_usdcToDai_differentReceiver() public assertAtomicPsmValueDoesNotChange {
+        _swapExactOutTest(usdc, dai, 100e18, 100e6, swapper, receiver);
+    }
+
+    function test_swapExactOut_usdcToSDai_differentReceiver() public assertAtomicPsmValueDoesNotChange {
+        _swapExactOutTest(usdc, sDai, 80e18, 100e6, swapper, receiver);
+    }
+
+    function testFuzz_swapExactOut_usdcToDai(
+        uint256 amountOut,
+        address fuzzSwapper,
+        address fuzzReceiver
+    ) public {
+        vm.assume(fuzzSwapper  != address(psm));
+        vm.assume(fuzzReceiver != address(psm));
+        vm.assume(fuzzReceiver != address(0));
+
+        amountOut = _bound(amountOut, 1, DAI_TOKEN_MAX);  // Zero amount reverts
+        uint256 amountIn = amountOut / 1e12;
```

Comment:
This seems wrong. Amount the user has to pay should not be rounding down from how much they get out. I think code needs to be modified with a `_divup` for the exact out case.